### ISSUE TITLE
Process excepts after restarting daemon

### ISSUE THREAD

**Issue by ahkole** [aiidateam/aiida-core#5882]
### Describe the bug

I recently updated my Python (from 3.9.13 to 3.10.9) and AiiDA installation (from 2.0.4 to 2.2.1) and now my processes fail
with an exception whenever I stop and restart the daemon while the process is running. The process report has the following traceback:

```
 | [3705|SiestaCalculation|on_except]: Traceback (most recent call last):
 |   File "<some path>/plumpy/processes.py", line 1218, in step
 |     next_state = await self._run_task(self._state.execute)
 |   File "<some path>/plumpy/processes.py", line 567, in _run_task
 |     result = await coro(*args, **kwargs)
 |   File "<some path>/aiida/engine/processes/calcjobs/tasks.py", line 482, in execute
 |     process_status = f'Waiting for transport task: {self._command}'
 | AttributeError: 'Waiting' object has no attribute '_command'
```

I'm not sure if the problem is related to me updating my environment, but I do know that before this I never had any problems with stopping and restarting the daemon while processes were running.

### Steps to reproduce

I haven't tried to reproduce it with a calculation which isn't a siesta calculation, but the exception doesn't sound
siesta specific, so I expect that you can probably reproduce it with any type of calculation (it definitely happens with the `SiestaBaseWorkchain` from the `aiida-siesta` plugin) with the following steps,
1. Start a process (in my case I submit the calculation to an external computer via SSH, I haven't tested it for local calculations)
2. Wait for the process state to change to `RUNNING`
3. Stop the AiiDA daemon
4. Wait for the calculation  to finish succesfullly
5. Restart the daemon
6. The process should now fail with the aforementioned exception

### Expected behavior

I expect the process to resume gracefully, detect that the calculation has finished in the meantime, retrieve the results and finish successfully.

### Your environment

- Operating system [e.g. Linux]: Arch Linux
- Python version [e.g. 3.7.1]: 3.10.9
- aiida-core version [e.g. 1.2.1]: 2.2.1
- aiida-siesta version: 2.0.1.dev0
- aiida-pseudo version: 0.9.0
- PostgreSQL version: 14.6
- RabbitMQ version: 3.10.7

**Comment by sphuber** [comment]
Does this also happen to any new processes that you launch now? Or only processes that were launched before you upgraded `aiida-core`. I am pretty sure it is probably the latter. Could you please confirm this?

Although we do guarantee an automatic migration of data and _completed_ processes between AiiDA versions, this is not the case for _active_ processes. Just as these can also not be exported, we cannot guarantee compatibility of active processes between AiiDA versions.

There is nothing we can do to "fix" this unfortunately. But any new jobs that you launch should work without issues. If that is _not_ the case. Please let us know

**Comment by ahkole** [comment]
@sphuber Thanks for getting back to me! There were no active processes when I upgraded so all processes for which this error occurs are processes that were launched after the upgrade. I am using old data nodes to launch the processes (i.e. the `Computer`, `Code` and multiple other `Input` data nodes were created before the upgrade), I don't know if that can also cause an issue?

**Comment by sphuber** [comment]
Interesting, thanks for confirming. Could you please paste the output of the command `pip freeze | grep -E 'aiida-core|plumpy'`

**Comment by ahkole** [comment]
Here it is:
```
❯ pip freeze | grep -E 'aiida-core|plumpy'
aiida-core==2.2.1
plumpy==0.21.3
```

**Comment by sphuber** [comment]
Thanks, I can now actually reproduce this. Seems like this bug was introduced in `v2.2.0` and snuck past our tests. I will look into fixing it a.s.a.p. For the time being, please down-grade `aiida-core` to `v2.1.2` which should not be affected.

**Comment by ahkole** [comment]
Okay, thanks! Good to know you are looking into it. Is there any way to migrate data created with a newer version of `aiida-core` to an older one? To downgrade I would have to restore a backup from before the database migration but this will not contain the results of calculations that I have run after the upgrade. Could you migrate data using an archive export and import? Or is it not possible to import archives created with newer versions of `aiida-core` into older versions?

**Comment by sphuber** [comment]
There weren't any database migrations between 2.1 and 2.2 so you should simply be able to run `pip install aiida-core==2.1.2` and then continue to use AiiDA.

**Comment by ahkole** [comment]
Great, thanks! I'll give it a try.

**Comment by sphuber** [comment]
Fixed in #5886 and released with v2.2.2
